### PR TITLE
appveyor: Use VS2017 for all our images

### PR DESCRIPTION
**Pull request by alexcrichton**
Originally added in #55935 to test build times, this was reverted
in #56201 due to a belief that it caused the exit code 259 spurious
errors. We've since learned, however, that the 259 exit code is likely
not related to this image update as we're getting it in a number of
locations now.

VS2017 looks like it may be required to compile LLVm in the near future,
notably discovered by #58408 where we attempted to update LLVM.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -1,4 +1,7 @@
 environment:
+  # This is required for at least an AArch64 compiler in one image, and is also
+  # going to soon be required for compiling LLVM.
+  APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017 Preview
```
Comment (ollie27) [inline appveyor.yml]: Does this still need to be the `Preview` image?

Comment (alexcrichton) [reply appveyor.yml]: I'm not sure, but I think it's best if we change one thing at a time

**Comment by kennytm** [comment]
@bors retry 

```
[02:13:42] warning: spurious network error (2 tries remaining): failed to get 200 response from `https://static.crates.io/crates/strsim/strsim-0.4.1.crate`, got 504
[02:13:42] warning: spurious network error (1 tries remaining): failed to get 200 response from `https://static.crates.io/crates/strsim/strsim-0.4.1.crate`, got 504
[02:13:42] error: failed to download from `https://crates.io/api/v1/crates/strsim/0.4.1/download`
[02:13:42] 
[02:13:42] Caused by:
[02:13:42]   failed to get 200 response from `https://static.crates.io/crates/strsim/strsim-0.4.1.crate`, got 504
```

**Pull request merged**